### ISSUE TITLE
Mismatch between the number of events requested and the actual number of events processed

### ISSUE THREAD

**Issue by gonzaponte** [next-exp/IC#270]
In at least a couple of cities, there is a block of code that makes things wrong. I am referring to:
```
nevt_in += 1
if self.max_events_reached(nevt_in):
    max_events_reached = True
    break
```
When using this block ***at the beginning of a loop***, it creates a shift of one unit between the number of events processed and `nevt_in`. This may be causing bugs to pass silently, particularly in tests, where, I think, we only process one event. In this case, we just do `nevt_in = 1` and right after we test whether `nevt_in == 1` and exit the loop without processing any event.

There are several solutions:
1. Move the nevt_in += 1 after the *if*.
2. Move the whole block to ***the end of the loop***, which seems more adequate to me.

Let's cross our fingers and hope the tests still pass!

**Comment by jacg** [comment]
> In at least a couple of cities, there is a block of code that makes things wrong.

I trust that, once we have finished refactoring the cities, this block (or whatever replaces it) will be written *exactly once* and re-used where appropriate!

**Comment by jjgomezcadenas** [comment]
that should be solved by #272
